cachelib/allocator/CacheAllocator.h

```diff
@@ -1980,41 +1985,65 @@ auto& mmContainer = getMMContainer(tid, pid, cid);
           mmContainer.remove(itr);
           candidates.push_back(candidate);
         } else {
-            ++itr;
+          ++itr;
         }
       }
     });
 
     for (Item *candidate : candidates) {
       auto evictedToNext = tryEvictToNextMemoryTier(*candidate, true /* from BgThread */);
       if (!evictedToNext) {
-	  auto token = createPutToken(*candidate);
```

Comment:
We should probably rebase to the upstream pretty soon and refactor this code into separate functions - it's getting pretty hard to review.